applications/GeoMechanicsApplication/custom_conditions/T_condition.cpp

```diff
@@ -0,0 +1,112 @@
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//
+//  Main authors:    Mohamed Nabi
+//                   John van Esch
+//
+
+#include "custom_conditions/T_condition.h"
+
+namespace Kratos {
+
+template <unsigned int TDim, unsigned int TNumNodes>
+TCondition<TDim, TNumNodes>::TCondition() : Condition()
+{
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+TCondition<TDim, TNumNodes>::TCondition(IndexType NewId, GeometryType::Pointer pGeometry)
+    : Condition(NewId, pGeometry)
+{
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+TCondition<TDim, TNumNodes>::TCondition(IndexType NewId,
+                                        GeometryType::Pointer pGeometry,
+                                        PropertiesType::Pointer pProperties)
+    : Condition(NewId, pGeometry, pProperties)
+{
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+TCondition<TDim, TNumNodes>::~TCondition() = default;
+
+template <unsigned int TDim, unsigned int TNumNodes>
+void TCondition<TDim, TNumNodes>::CalculateLocalSystem(MatrixType& rLeftHandSideMatrix,
+                                                       VectorType& rRightHandSideVector,
+                                                       const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    if (rLeftHandSideMatrix.size1() != TNumNodes || rLeftHandSideMatrix.size2() != TNumNodes) {
+        rLeftHandSideMatrix.resize(TNumNodes, TNumNodes, false);
+    }
+    noalias(rLeftHandSideMatrix) = ZeroMatrix(TNumNodes, TNumNodes);
+
+    if (rRightHandSideVector.size() != TNumNodes) {
+        rRightHandSideVector.resize(TNumNodes, false);
+    }
+    noalias(rRightHandSideVector) = ZeroVector(TNumNodes);
+
+    CalculateAll(rLeftHandSideMatrix, rRightHandSideVector, rCurrentProcessInfo);
+
+    KRATOS_CATCH("")
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+void TCondition<TDim, TNumNodes>::EquationIdVector(EquationIdVectorType& rResult,
+                                                   const ProcessInfo& rCurrentProcessInfo) const
+{
+    KRATOS_TRY
+
+    if (rResult.size() != TNumNodes) {
+        rResult.resize(TNumNodes, false);
+    }
+
+    const GeometryType& rGeom = GetGeometry();
+    for (unsigned int i = 0; i < TNumNodes; ++i) {
+        rResult[i] = rGeom[i].GetDof(TEMPERATURE).EquationId();
+    }
+
+    KRATOS_CATCH("")
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+void TCondition<TDim, TNumNodes>::CalculateAll(MatrixType& rLeftHandSideMatrix,
+                                               VectorType& rRightHandSideVector,
+                                               const ProcessInfo& rCurrentProcessInfo)
+{
+    CalculateRHS(rRightHandSideVector, rCurrentProcessInfo);
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+void TCondition<TDim, TNumNodes>::CalculateRHS(VectorType& rRightHandSideVector,
```

Comment:
I think it would be nice to make this function purely virtual (= 0) such that we enforce that the function is implemented in a derived class, instead of throwing this error run-time

Reply:
Unfortunately, there are a few instances of this class and this does not work with the pure virtual function.

Reply:
Okay, if we need instantiate TCondition somewhere as a concrete class that's indeed not possible